master/buildbot/changes/p4poller.py

```diff
@@ -235,6 +235,16 @@ def _poll(self):
             args.extend(['-m', '1', '%s...' % (self.p4base,)])
 
         result = yield self._get_process_output(args)
+        # decode the result from its designated encoding
+        try:
+            result = bytes2unicode(result, self.encoding)
```

Comment:
Having the same pattern in the file is not always a good reason.
What will happen here is that the change is just ignored. I think it is better, as an overall best practice to be more gentler when there are errors. Try our best to still injest the change even if it looks suspicious instead of just rejecting it.

I think mode="replace" is good at that.

maybe we should do:

```
try:
    result = bytes2unicode(result, self.encoding)
except UnicodeError:
   log.msg("Warning: cannot fully decode {} in {}".format(result, self.encoding))
   result = bytes2unicode(result, self.encoding, mode="replace")
```